lib/internal/http2.js

```diff
@@ -1605,20 +1606,7 @@ class Http2ClientRequest extends Http2Outgoing {
       const _handle = this.stream.session.request(mapToHeaders(this[kHeaders]), true);
       if (_handle instanceof http2.Http2Stream) {
         this[kId] = _handle.getId();
-        this.stream.once('handle', () => {
-          if (this[kTrailers] instanceof Map) {
-            for (const v of this[kTrailers]) {
-              const key = String(v[0]);
-              const value = v[1];
-              if (Array.isArray(value) && value.length > 0) {
-                for (const item of value)
-                  this.stream.addTrailer(key, String(item));
-              } else {
-                this.stream.addTrailer(key, String(value));
-              }
-            }
-          }
-        });
+        this.stream.once('handle', kBeginSendHandle(this))
```

Comment:
the function declaration is hoisted. It happens even if the if statement is false.